src/MyTested.AspNetCore.Mvc.Controllers.Attributes/Builders/Attributes/AuthorizeAttributeTestBuilder.cs

```diff
@@ -0,0 +1,142 @@
+﻿namespace MyTested.AspNetCore.Mvc.Builders.Attributes
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Contracts.Attributes;
+    using Internal.TestContexts;
+    using Microsoft.AspNetCore.Authorization;
+    using Utilities.Extensions;
+
+    /// <summary>
+    /// Used for testing <see cref="AuthorizeAttribute"/>.
+    /// </summary>
+    public class AuthorizeAttributeTestBuilder : BaseAttributeTestBuilder<AuthorizeAttribute>,
+        IAndAuthorizeAttributeTestBuilder
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthorizeAttributeTestBuilder"/> class.
+        /// </summary>
+        /// <param name="testContext"><see cref="ComponentTestContext"/> containing data about the currently executed assertion chain.</param>
+        /// <param name="failedValidationAction">Action to call in case of failed validation.</param>
+        public AuthorizeAttributeTestBuilder(
+            ComponentTestContext testContext,
+            Action<string, string> failedValidationAction)
+            : base(testContext, nameof(AuthorizeAttribute), failedValidationAction)
+            => this.Attribute = new AuthorizeAttribute();
+
+        /// <inheritdoc />
+        public IAndAuthorizeAttributeTestBuilder WithPolicy(string policy)
+        {
+            if (!string.IsNullOrEmpty(policy))
+            {
+                this.Attribute = new AuthorizeAttribute(policy)
+                {
+                    Roles = this.Attribute.Roles,
+                    AuthenticationSchemes = this.Attribute.AuthenticationSchemes
+                };
+
+                this.Validations.Add((expected, actual) =>
+                {
+                    var expectedPolicy = expected.Policy;
+                    var actualPolicy = actual.Policy;
+
+                    if (!string.Equals(expectedPolicy, actualPolicy, StringComparison.OrdinalIgnoreCase))
+                    {
+                        this.FailedValidationAction(
+                            $"{this.ExceptionMessagePrefix}'{expectedPolicy}' policy",
+                            $"in fact found '{actualPolicy}'");
+                    }
+                });
+            }
+
+            return this;
+        }
+
+        /// <inheritdoc />
+        public IAndAuthorizeAttributeTestBuilder WithRole(string role)
+        {
+            if (!string.IsNullOrEmpty(role))
+            {
+                this.Attribute.Roles = role;
+
+                this.Validations.Add((expected, actual) =>
+                {
+                    var expectedRoles = expected.Roles;
+                    var actualRoles = actual.Roles;
+
+                    if (!string.Equals(expectedRoles, actualRoles, StringComparison.OrdinalIgnoreCase))
```

Comment:
This logic will fail, if you have `Authorize(Roles = "Administrator,Moderator")` and you check like this `WithRole("Administrator")`. The test should pass, but it will not.